src/block.ts

```diff
@@ -0,0 +1,381 @@
+    displayCredit ? `${caption} ${credit}` : caption;
+
+interface ImageProps {
+    url: string;
+    alt: string;
+    salt: string;
+    caption: string;
+    displayCredit: boolean;
+    credit: string;
+}
+
+const imageStyles = css`
+    margin-left: ${basePx(-1)};
+    margin-right: ${basePx(-1)};
+
+    ${from.phablet} {
+        margin-left: 0;
+        margin-right: 0;
+    }
+
+    ${from.wide} {
+        margin: 1em 0;
+    }
+
+    img {
+        width: 100%; 
+    }
+
+    figcaption {
+        font-size: 1.4rem;
+        line-height: 1.8rem;
+        color: ${palette.neutral[46]};
+        ${textSans}
+
+        ${until.phablet} {
+            padding-left: ${basePx(1)};
+            padding-right: ${basePx(1)};
+        }
+    }
+`;
+
+const Image = ({ url, alt, salt, caption, displayCredit, credit }: ImageProps): ReactElement =>
+    styledH('figure', { css: imageStyles },
+        h('img', {
+            sizes: '100%',
+            srcSet: srcset(salt)(url),
+            alt,
+            src: transformUrl(salt, url, 500),
+        }),
+        h('figcaption', null, makeCaption(caption, displayCredit, credit)),
+    );
+
+const pullquoteStyles = (colour: string): SerializedStyles => css`
+    font-weight: 200;
+    font-size: 2.2rem;
+    line-height: 1.3;
+    color: ${colour};
+    ${headlineFont}
+    margin: 0;
+
+    p {
+        margin: 1em 0;
+
+        &::before {
+            ${icons}
+            font-size: 2.2rem;
+            content: '\\e11c';
+            display: inline-block;
+            margin-right: ${basePx(1)};
+        }
+    }
+
+    footer {
+        font-size: 1.8rem;
+        margin-top: 4px;
+
+        cite {
+            font-style: normal;
+        }
+    }
+`;
+
+type PullquoteProps = {
+    quote: string;
+    attribution: Option<string>;
+    pillar: Pillar;
+};
+
+const Pullquote = (props: PullquoteProps): ReactElement =>
+    styledH('aside', { css: pullquoteStyles(getPillarStyles(props.pillar).kicker) },
+        h('blockquote', null,
+            h('p', null, props.quote)
+        )
+    );
+
+const richLinkWidth = '13.75rem';
+
+const richLinkStyles = css`
+    background: ${palette.neutral[97]};
+    padding: ${basePx(1)};
+
+    h1 {
+        margin: 0;
+    }
+
+    p {
+        margin: ${basePx(1, 0)};
+    }
+
+    span {
+        display: none;
+    }
+
+    a {
+        text-decoration: none;
+    }
+
+    float: left;
+    clear: left;
+    width: ${richLinkWidth};
+    margin: ${basePx(1, 2, 1, 0)};
+
+    ${from.wide} {
+        margin-left: calc(-${richLinkWidth} - 16px - 24px);
+    }
+
+    ${darkModeCss`
+        border-top: 1px solid ${palette.neutral[60]};
+        border-bottom: 1px solid ${palette.neutral[60]};
+        a {
+            &::before {
+                color: ${palette.neutral[60]};
+            }
+        }
+    `}
+`;
+
+const RichLink = (props: { url: string; linkText: string; pillar: Pillar }): ReactElement =>
```

Comment:
This file isn't too large right now but if we have to handle more types in the future maybe we should pull them out into `blocks/components` or somewhere?

Reply:
Might make it easier to use them as storybook components too.

Reply:
Yeah that might make sense. I'm trying to write everything in one file and then split it into smaller files when I discover from the code what those smaller files should be.